src/index.ts

```diff
@@ -180,7 +180,7 @@ export function createBirpc<RemoteFunctions = Record<string, never>, LocalFuncti
 
           if (timeout >= 0) {
             timeoutId = setTimeout(() => {
-              reject(new Error(`[birpc] timeout on calling "${method}"`))
+              reject(new Error(`[birpc] timeout on calling "${method}" with arguments "${serialize(args)}"`))
```

Comment:
I am afraid that if the serialization could create another error or get infinite looped. Should we add one extra safety here?

Reply:
`serialize` is called in this scope before this timeout could happen. So if it would throw error, I think it would happen before this `Error` is created.

But do we actually even need to call `serialize`? This `Error` is created on the caller's side. Serializing it could make it non-readable. Maybe just stringify it? 

This is what would happen with `v8.serialize`:

```js
> v8.serialize('some-module', 'ssr')
<Buffer ff 0f 22 0b 73 6f 6d 65 2d 6d 6f 64 75 6c 65>

> new Error(`Testing: ${v8.serialize('some-module', 'ssr')}`)
Error: Testing: �"
                  some-module
    at REPL16:1:1
    at Script.runInThisContext (node:vm:123:12)
    ```

Reply:
Yeah I think it's more like stringify it. Maybe we could do a simple `args.map(i => String(i))` to display literals but ignore nested objects

Reply:
> do a simple `args.map(i => String(i))` to display literals but ignore nested objects

Should we leave `[object Object]` args in the output or filter them out?

```js
> new Error(`Testing ${['/server.ts', { transformMode: 'ssr'}].map(arg => String(arg))}`)
Error: Testing /server.ts,[object Object]
```

Or how about adding a small formatter function that would format objects just one level deep. Is this over-engineering? :smile:

```js
function stringifyArg(arg) {
  const string = String(arg);
  if (string !== "[object Object]") return string;

  const oneLevel = Object.entries(arg).reduce(
    (all, [name, value]) => ({ ...all, [name]: String(value) }),
    {}
  );
  return JSON.stringify(oneLevel);
}
```

```js
const args = [
  "some-module",
  "ssr",
  { filepath: "./some-module.mjs", deeply: { nested: { object: { should: { be: { ignored: true } } } } } },
  ['some', 'more', 'options'],
]

new Error(`[birpc] timeout on calling "${method}" with arguments "${args.map(stringifyArg)}"`);
> Error: [birpc] timeout on calling "fetch" with arguments "some-module,ssr,{"filepath":"./some-module.mjs","deeply":"[object Object]"},some,more,options"

```

Reply:
Both sound ok to me 👌